openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -161,24 +161,6 @@ def can_show_upgrade_sock(self):
     def license(self):
         return self.course.license
 
-    @property
-    def username(self):
-        return self.effective_user.username
-
-    @property
-    def course_access(self) -> dict:
```

Comment:
The alternative version of this in courseHome metadata does not have the same access error. Is it needed? https://github.com/openedx/edx-platform/blob/ae618151a96a20b4435bfadf50dc47f784fa90ea/lms/djangoapps/course_home_api/course_metadata/views.py#L133

Reply:
The `CoursewareMicrofrontendDisabledAccessError` is just one way of returning the data. `load_access.to_json()` is another way. The implementation here is just specifying a return for the specific use case of the courseware side of the learning MFE not being visible. So needed? No. But if you find it nice, then it's totally fine to leave.

The bigger question here in my opinion is the difference between how the course home and courseware handle the mfe being enabled. Here, they return a 200 with the `CoursewareMicrofrontendDisabledAccessError`. In the course home side of things, if the legacy view is active, we 404 on our endpoint and then redirect and handle it that way. I think we should likely be consistent between these.